### ISSUE TITLE
All domains to community

### ISSUE THREAD

**Issue by markus2330** [ElektraInitiative/libelektra#1171]
Goal: have everything from the Elektra project on the community server, reachable by https:

- [ ] tree.libelektra.org, blob.libelektra.org, master.libelektra.org (DNS for master already pointing to community), should redirect to github tree
- [ ] build.libelektra.org -> redirect to 8080
- [X] restapi.libelektra.org
- [X] community.libelektra.org, web.libelektra.org
- [ ] www.libelektra.org
- [ ] doc.libelektra.org

**Comment by markus2330** [comment]
github.libelektra.org

**Comment by Namoshek** [comment]
Side note: We should use https://raw.githubusercontent.com/ElektraInitiative/libelektra/master/ for the last one directly as github will redirect a second time otherwise.

**Comment by markus2330** [comment]
Thanks, you are right! I updated it. Can you add these domains in Apache Config?

**Comment by Namoshek** [comment]
Are they routed to the community server already?

**Comment by Namoshek** [comment]
Where should doc.libelektra.org point to?

I added the rest already, www.libleketra.org currently redirects to community.libelektra.org, we can change this when we go live. All sites http + https.

**Comment by markus2330** [comment]
Thank you, great work!

> Are they routed to the community server already?

I now rerouted tree, blob, bugs, git, github, raw, and source.

Can you write a overview of what is available and where it points to? (Best would be to start with a doc/README.md with an overview where all the docu is)

Btw. Where is the main README.md linked to at the home page?

doc.libelektra.org also needs changes in the build job (publish to homepage instead of some other server). If this is done, doc.libelektra.org can point to the same server (community.libelektra.org).

**Comment by Namoshek** [comment]
> Btw. Where is the main README.md linked to at the home page?

Nowhere so far.

> doc.libelektra.org also needs changes in the build job (publish to homepage instead of some other server). If this is done, doc.libelektra.org can point to the same server (community.libelektra.org).

Hm ok, but we should fix the homepage build first.

**Comment by markus2330** [comment]
> Nowhere so far.

We could also get the rest of its content into the homepage (in particular how to get started in contributing to Elektra) or include it as overview over the project (Documentation -> Overview)?

> Hm ok, but we should fix the homepage build first.

Definitely, the migration would be nice (for https), but should not hinder the progress of the website going life.

**Comment by Namoshek** [comment]
Done?

**Comment by markus2330** [comment]
Yes, done from your side.

Let us keep the ticket open: Maybe there are some build jobs not working.

**Comment by markus2330** [comment]
I updated build.debian.org/debian to work with the new setup.

jenkins and the other apache (for debian packages) are now only listening to localhost only.

**Comment by Namoshek** [comment]
https://build.libelektra.org/debian does now redirect to http://localhost:8082/debian/

**Comment by markus2330** [comment]
Thanks for reporting! I was trying to proxy through the wordpress site, but it seems to be not worth the effort.

**Comment by markus2330** [comment]
Added ProxyPassReverse, too. Should work now.

@manuelm Can you add https for 194.117.254.29/elektra-stable/ (DNS: debian-stable.libelektra.org)?

**Comment by Namoshek** [comment]
Getting this on my server (tried there as second check to avoid DNS caching issue):
```
$> curl https://build.libelektra.org/debian
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>301 Moved Permanently</title>
</head><body>
<h1>Moved Permanently</h1>
<p>The document has moved <a href="http://localhost:8082/debian/">here</a>.</p>
<hr>
<address>Apache/2.2.22 (Debian) Server at localhost Port 8082</address>
</body></html>
```

**Comment by markus2330** [comment]
Yes, of course. The ProxyPassReverse only modifies the `Location` header, but not the html content.

```
curl -I "http://build.libelektra.org/debian"
HTTP/1.1 301 Moved Permanently
Date: Wed, 21 Dec 2016 14:23:21 GMT
Server: Apache/2.4.10 (Debian)
Location: https://build.libelektra.org/debian
Connection: close
Content-Type: text/html; charset=iso-8859-1
```

**Comment by Namoshek** [comment]
Ok, wget seems to retrieve the correct file now.

**Comment by manuelm** [comment]
> @manuelm Can you add https for 194.117.254.29/elektra-stable/ (DNS: debian-stable.libelektra.org)?

done

**Comment by markus2330** [comment]
Thank you!
